alacritty_terminal/src/term/mod.rs

```diff
@@ -1718,8 +1718,33 @@ impl<T: EventListener> ansi::Handler for Term<T> {
 
     /// Set the clipboard
     #[inline]
-    fn set_clipboard(&mut self, string: &str) {
-        self.clipboard.store(ClipboardType::Clipboard, string);
+    fn set_clipboard(&mut self, clipboard: u8, base64: &[u8]) {
+        let clipboard_type = match clipboard {
+            b'c' => ClipboardType::Clipboard,
+            b'p' | b's' => ClipboardType::Selection,
+            _ => return,
+        };
+
+        if let Ok(bytes) = base64::decode(base64) {
+            if let Ok(text) = str::from_utf8(&bytes) {
+                self.clipboard.store(clipboard_type, text);
+            }
+        }
+    }
+
+    /// Write clipboard data to child.
+    #[inline]
+    fn write_clipboard<W: io::Write>(&mut self, clipboard: u8, writer: &mut W) {
+        let clipboard_type = match clipboard {
+            b'c' => ClipboardType::Clipboard,
+            b'p' | b's' => ClipboardType::Selection,
+            _ => return,
+        };
+
+        let text = self.clipboard.load(clipboard_type);
+        let base64 = base64::encode(&text);
+        let escape = format!("\x1b]52;{};{}\x07", clipboard as char, base64);
```

Comment:
I wonder how we should map this behavior to upcoming clipboard rework(using request_load). Do you think that providing optional callback would make sense for such thing?

Reply:
This is actually a very good question. Currently we just have access to a `writer` which is direct mutable access to the PTY's fd.

However we already have things in place for writing to PTY on callback, since our `write_to_pty` uses channels. So I think we'd just have to pass the `event_proxy` and then have it write directly to PTY itself, the same way we load from clipboard when pressing the key combinations.

So I don't think this should be any problem! Especially since our `Term` already has an `event_proxy` available.

Reply:
The problem is not to write to a pty, the problem is to write a correct string. So I'm talking about optional transform for a loaded text `|String| -> String` lambda, since our async clipboard loader can write_to_pty, however it can't transform loaded text. For bracketed paste we're checking terminal mode, which is fine, however for this thing, we don't have any mode (and I don't want to add one for it).

Reply:
Oh, right. Since we need to do more than just load the text.

I think we might want to rethink our loading strategy completely tbh. Automatically writing the output to the event loop seems like a bad idea anyways, we should have something more like this:

```rust
clipboard.load(|data| {
    event_proxy.send(Paste(data));
});
```

Having the clipboard data do more than just load the clipboard seems like a strange design decesion to me anyways, even though I haven't thought about it before. Having a callback that is executed once the data is present makes a lot more sense.

Reply:
I also like this strategy, but I'm worry about carrying event_proxy around, anyway that's enough info to work with for now. Thx.